specification/cognitiveservices/OpenAI.Inference/models/images.tsp

```diff
@@ -0,0 +1,137 @@
+import "@azure-tools/typespec-azure-core";
+import "@typespec/http";
+
+using Azure.Core;
+using TypeSpec.Http;
+
+namespace Azure.OpenAI;
+
+@doc("The desired size of the generated images. Must be one of 256x256, 512x512, or 1024x1024.")
+enum ImageSize {
+  @doc("Small image size (256x256).")
+  Small: "256x256",
+
+  @doc("Medium image size (512x512).")
+  Medium: "512x512",
+
+  @doc("Large image size (1024x1024).")
+  Large: "1024x1024",
+}
+
+@doc("Represents the request data used to generate images.")
+model ImageRequest {
+  @doc("A description of the desired images.")
+  prompt: string;
+
+  @doc("The number of images to generate (defaults to 1).")
+  n?: int32;
+
+  @doc("The desired size of the generated images. Must be one of 256x256, 512x512, or 1024x1024 (defaults to 1024x1024).")
+  size?: ImageSize;
+
+  @doc("A unique identifier representing your end-user, which can help to monitor and detect abuse.")
+  user?: string;
+}
+
+@doc("The image url if successful, and an error otherwise.")
+model ImageResult {
```

Comment:
This model looks incomplete. What happens if you specify `response_format = "b64_json"`? Or have we explicitly decided not to support that?

Reply:
response_format is not part of the service API definition we based these on

Reply:
I don't know when it was added, but it is at least supported in OAI...

Reply:
yeah, for AOAI it is not there in the latest service definitions, we can add it now or when we do the work to support the vanilla (non-LRO) generation API

Reply:
Since we know it is coming, we should at least spend a couple of minutes to make sure we are not digging ourselves into a hole w. regards to structure or naming. E.g. maybe we want to have an ImageLocation rather than Image name for the returned model if we expect to provide differently named methods for the b64 version and the url version.

Reply:
Looking at the OAI swagger, it seems that the change to support the b64 format would be compatible with the current structure. When that comes, I imagine the model would be close to this:

```
model ImageResult {
  url?: url,
  b64_json?: string
}
```

Reply:
The proposal above misses two important pieces of information; they are mutually exclusive, and exactly one of the properties must be provided. A more precise description (names subject to change) is:

```
model ImageLocation {
    url: url;
}

model ImageData {
    b64_json: string;
}

alias ImageResult = ImageData | ImageLocation;
```

Reply:
that is fair, that'd be an improvement over how OpenAI defined their API. I'll for now rename ImageResult to ImageLocation. When/if we use the union we may need to figure out how to make that work with the openapi generation requirement the repo has.

I'll also ask to the service if they plan to ever support base64 output for the LRO operation.

Reply:
Yeah, openapi2 didn't have a good way to express this. In OpenAPI 3+, it would be a `oneOf`. The closest thing we have in autorest/azure swagger land is to make it an "overload" (using the x-ms-paths vendor extension in our swagger) and have two different response models. Which is a pain (to say the least).